grid_map_visualization/src/visualizations/FlatPointCloudVisualization.cpp

```diff
@@ -6,52 +6,61 @@
  *   Institute: ETH Zurich, ANYbotics
  */
 
-#include <grid_map_visualization/visualizations/FlatPointCloudVisualization.hpp>
 #include <grid_map_ros/GridMapRosConverter.hpp>
 
-#include <sensor_msgs/PointCloud2.h>
+#include <sensor_msgs/msg/point_cloud2.hpp>
 
-namespace grid_map_visualization {
+#include <string>
 
-FlatPointCloudVisualization::FlatPointCloudVisualization(ros::NodeHandle& nodeHandle, const std::string& name)
-    : VisualizationBase(nodeHandle, name),
-      height_(0.0)
+#include "grid_map_visualization/visualizations/FlatPointCloudVisualization.hpp"
+
+namespace grid_map_visualization
+{
+
+FlatPointCloudVisualization::FlatPointCloudVisualization(
+  rclcpp::Node::SharedPtr node,
+  const std::string & name)
+: VisualizationBase(node, name),
+  height_(0.0)
 {
 }
 
 FlatPointCloudVisualization::~FlatPointCloudVisualization()
 {
 }
 
-bool FlatPointCloudVisualization::readParameters(XmlRpc::XmlRpcValue& config)
+bool FlatPointCloudVisualization::readParameters()
 {
-  VisualizationBase::readParameters(config);
-
   height_ = 0.0;
-  if (!getParam("height", height_)) {
-    ROS_INFO("FlatPointCloudVisualization with name '%s' did not find a 'height' parameter. Using default.", name_.c_str());
+  node_->declare_parameter(name_ + ".params.height", 0.0);
+  if (!node_->get_parameter(name_ + ".params.height", height_)) {
+    RCLCPP_INFO(
+      node_->get_logger(),
+      "FlatPointCloudVisualization with name '%s' "
+      "did not find a 'height' parameter. Using default.",
+      name_.c_str());
   }
 
   return true;
 }
 
 bool FlatPointCloudVisualization::initialize()
 {
-  publisher_ = nodeHandle_.advertise<sensor_msgs::PointCloud2>(name_, 1, true);
```

Comment:
Just noticed the publishers were latched, do we need them latched?

Reply:
Good q- Max?

Reply:
If possible yes, for visualization having it latched is nice.